application/legacy/LegacyLinkDB.php

```diff
@@ -188,14 +188,15 @@ public function offsetGet($offset)
     /**
      * Iterator - Returns the current element
      */
-    public function current()
+    public function current(): array
     {
         return $this[$this->keys[$this->position]];
     }
 
     /**
      * Iterator - Returns the key of the current element
      */
```

Comment:
Use docblock to document return type until we can enforce it at the language level

```suggestion
     * Iterator - Returns the key of the current element
     *
     * @return int|string
     */
```